### PR TITLE
[docs] add docstrings in `pipline_stable_diffusion.py`

### PR DESCRIPTION
**Pull request by jeongiin**
# What does this PR do?

Fixes : #9567 

I have added a docstring to the `rescale_noise_cfg` function. I plan to continue addressing this issue, and I would greatly appreciate any advice you could offer on writing effective commit messages or PRs. Thank you!

ps. I've worked on other tasks as well, so there are many commit messages, but only one file was actually changed.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@a-r-r-o-w @stevhliu

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 1]
```diff
@@ -65,9 +65,22 @@
...
 
 def rescale_noise_cfg(noise_cfg, noise_pred_text, guidance_rescale=0.0):
-    """
-    Rescale `noise_cfg` according to `guidance_rescale`. Based on findings of [Common Diffusion Noise Schedules and
-    Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf). See Section 3.4
+    r"""
+
+    Rescales `noise_cfg` tensor based on `guidance_rescale` to improve image quality and fix overexposure.
+    Based on findings of [Common Diffusion Noise Schedules and Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf). 
+    See Section 3.4.
```
Comment (stevhliu) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:72]:
```suggestion
    Based on Section 3.4 from [Common Diffusion Noise Schedules and Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf). 
```

**Review by stevhliu — APPROVED** [review]
Nice job!

**Comment by a-r-r-o-w** [comment]
Hi @jeongiin, just need to run `make style` and push changes here to be ready for merge I think

**Comment by jeongiin** [comment]
Hello @a-r-r-o-w ! I fixed style!

**Comment by jeongiin** [comment]
Hello @a-r-r-o-w and @stevhliu  I fixed style!

**Comment by stevhliu** [comment]
Thanks! Can you also run `make fix-copies` to align all the `# Copied from` statements?

**Comment by jeongiin** [comment]
Thanks for catching that!! @stevhliu

**Comment by stevhliu** [comment]
Hmm, the copy inconsistencies are still there for some reason. Can you try running `python utils/check_copies.py --fix_and_overwrite` instead?

**Comment by jeongiin** [comment]
I've tried running `python utils/check_copies.py --fix_and_overwrite`, but nothing happens to the problematic file. Could you suggest another solution? :(

**Comment by stevhliu** [comment]
Hmm, I'm not entirely sure then! Running those commands and pushing their changes should fix the tests. @sayakpaul do you have another suggestion?

**Comment by sayakpaul** [comment]
We need to apply `make fix-copies` to fix the errors.

**Comment by a-r-r-o-w** [comment]
@stevhliu @sayakpaul good to merge?

**Comment by sayakpaul** [comment]
Okay for me!

**Comment by HanLiii** [comment]
Will pipeline_animatediff_sdxl_video2video.py be released in the future?

**Comment by a-r-r-o-w** [comment]
Hi @HanLiii, we plan to make several improvements with #9672, which should allow creating arbitrary pipelines easily. We're ideating about the best way to move forward with the design that would allow this, so if you have any suggestions regarding that, it'd be awesome. For now, I think everyone on the team is occupied so we will not be able to commit to creating such a pipeline. If you would be interested in contributing a pipeline for it, we'd be happy to review and merge :)

**Pull request merged**